Docs/BazelForiOSDevelopers.md

```diff
@@ -0,0 +1,522 @@
+documentation for all possible arguments. _Note: Unlike Starlark rules which
+are easily added on to Bazel, the `objc_library` is part of the internal java
+rules shipped with the Bazel
+binary._ 
+
+_In the Xcode world, configuration is defined within Xcode projects and
+proprietary xcconfigs. In Bazel it's easy to establish norms, enforce
+consistency with abstractions like `objc_library`, and refactor_
+
+### Aspects
+
+Aspects are another extension point in Bazel. Simply put, they allow the
+developer to traverse the build graph and collect information or generate
+actions on the way. 
+
+> Aspects are a feature of Bazel that are basically like fan-fic, if build rules
+were stories: aspects let you add features that require intimate knowledge of
+the build graph, but that that the rule maintainer would never want to add.
+
+_- [Kristina Chodorow, Aspects: the fan-fic of build rules](https://kchodorow.com/2017/01/10/aspects-the-fan-fic-of-build-rules/)_
+
+Combined with Rules and the Bazel command line, the user to create robust
+architectures and powerful abstractions. Like rules, generally, defining custom
+rules isn't required but can improve and consolidate functionality. For more
+information about aspects, see [Aspects the fan-fic of build
+rules](https://kchodorow.com/2017/01/10/aspects-the-fan-fic-of-build-rules/)
+
+### Toolchains
+
+In addition to configuring the build with rules, Bazel provides an additional
+primitive, the toolchain. Toolchains provide default compilers and arguments
+for those compilers. For the native c++ rules, this is relied upon to configure
+the many flags required for cross compilation.
+
+For most iOS projects, configuring the c++ toolchain isn't required. Defining a
+custom toolchain is useful to gives full control over compiler invocations or
+to customize the compiler being invoked.  Please see the [toolchain
+documentation](https://docs.bazel.build/versions/master/tutorial/cc-toolchain-config.html)
+to learn more.
+
+### Generated Xcode projects
+
+In an Xcode world, folks checked in a project which contains an listing of
+files, build settings, and IDE state. When the project scales, the project file
+model breaks down. Auditing and code reviewing config changes in Xcode projects
+quickly becomes difficult. With Bazel, human readable `BUILD` files are the
+source of truth for build and Xcode configuration. Tools like
+[XCHammer](https://github.com/pinterest/xchammer), and
+[Tulsi](https://github.com/bazelbuild/tulsi) use an aspect to traverse the
+build graph and extract metadata required to generate a project.  These tools
+make it easier to manage the project and generate on demand - not needing to
+check it in.
+
+XCHammer provides a rule to [bazel build Xcode
+projects](ihttps://github.com/pinterest/xchammer#bazel-build-xcode-projects).
+Simply declare the rule with the project and Bazel build the target.
+```
+load("@xchammer_resources//:xcodeproject.bzl", "xcode_project")
```

Comment:
Does this require declaring git_repository for xchammer_resources in the WORKSPACE file?

Reply:
In Bazel you need to load every symbol - the XCHammer README also has the `load` statement to add this https://github.com/pinterest/xchammer/pull/235/files#diff-af23f534f9648f42a375a8dbf3792f8bR122